src/main/java/slave/elements/Parser.java

```diff
@@ -0,0 +1,139 @@
+package slave.elements;
+
+import slave.command.AddToDoCommand;
+import slave.command.NullCommand;
+import slave.command.ListCommand;
+import slave.command.HelpCommand;
+import slave.command.ExitCommand;
+import slave.command.DoneCommand;
+import slave.command.DeleteCommand;
+import slave.command.ClearCommand;
+import slave.command.AddEventCommand;
+import slave.command.Command;
+import slave.command.AddDeadlineCommand;
+import slave.command.FindCommand;
+
+import slave.exception.DukeException;
+import slave.exception.MissingDateException;
+import slave.exception.MissingDescriptionException;
+import slave.exception.MissingTaskException;
+
+/**
+ * A parser to take in input by user and returns the appropriate command.
+ */
+public class Parser {
+
+    /**
+     * Static method which parses input by user and returns the corresponding command object.
+     *
+     * @param fullCommand Input by user.
+     * @return Corresponding command based on user input.
+     * @throws DukeException For invalid input.
+     */
+    public static Command parse(String fullCommand) throws DukeException {
+        String[] tokens = fullCommand.split(" ");
+        String firstWord = tokens[0];
+        switch (firstWord) {
+        case "bye":
+            return new ExitCommand();
+        case "list":
+            return new ListCommand();
+        case "help":
+            return new HelpCommand();
+        case "clear":
+            return new ClearCommand();
+        case "find":
+            return new FindCommand(fullCommand.substring(5));
+        case "done":
+            checkValidity("done", fullCommand, tokens);
+            return new DoneCommand(Integer.parseInt(tokens[1]));
+        case "delete":
+            checkValidity("delete", fullCommand, tokens);
+            return new DeleteCommand(Integer.parseInt(tokens[1]));
+        case "todo":
```

Comment:
This method may seem a bit lengthy, would it be better to abstract the logic of adding the different tasks into an AddCommand class, which would then have your AddToDoCommand, AddDeadlineCommand and AddEventCommand extend from? It can potentially be helpful since both deadlines and events require formatting of the date.